File: python3_solution_set2/2399. Check Distances Between Same Letters.py
class Solution:
    def checkDistances(self, s: str, dist: List[int]) -> bool:
        
        g = {}
        for x in range(len(s)):
            if s[x] not in g:
                g[s[x]] = x
                
            else:
                g[s[x]] = x - g[s[x]] - 1
        
        for x in g:
            if g[x]!=dist[ord(x)-ord('a')]:
                return False
            
        return True
        
        

File: cpp_solutions/2399. Check Distances Between Same Letters.cc
class Solution {
public:
    void PrintMap(map<char,int> g) {
        for(auto it : g) {
            cout << it.first << "-" << it.second;
            cout << "\n";
        }
    }
    
    bool checkDistances(string s, vector<int>& distance) {
        map<char,int> g;
        
        for(int x=0; x<s.length(); ++x) {
            if (g.find(s.at(x))==g.end()) {
                g.insert({s.at(x),x});
            } else {
                int temp = g[s.at(x)];
                g[s.at(x)] = x - temp - 1;
            }
        }
        // PrintMap(g);
        for(auto it : g) {
            if(it.second!=distance[int(it.first)-int('a')]) {
                return false;
            }
        }
        
        return true;
    }
};
